Jenkinsfile

```diff
@@ -0,0 +1,335 @@
+              unstash 'source'
+              dir("${BASE_DIR}"){    
+                sh """#!/bin/bash
+                ./scripts/jenkins/test.sh
+                """
+                codecov('apm-agent-go')
+              }
+            }
+          }
+          post { 
+            always {
+              coverageReport("${BASE_DIR}/build")
+              junit(allowEmptyResults: true, 
+                keepLongStdio: true, 
+                testResults: "${BASE_DIR}/build/junit-*.xml")
+            }
+          }
+        }
+        stage('Benchmarks') {
+          agent { label 'linux' }
+          environment {
+            PATH = "${env.PATH}:${env.HUDSON_HOME}/go/bin/:${env.WORKSPACE}/bin"
+            GOPATH = "${env.WORKSPACE}"
+          }
+          
+          when { 
+            beforeAgent true
+            allOf { 
+              //branch 'master';
+              environment name: 'bench_ci', value: 'true' 
+            }
+          }
+          steps {
+            withEnvWrapper() {
+              unstash 'source'
+              dir("${BASE_DIR}"){    
+                sh """#!/bin/bash
+                ./scripts/jenkins/bench.sh
+                """
+                sendBenchmarks(file: 'build/bench.out')
+              }
+            }
+          } 
+          post {
+            always {
+              junit(allowEmptyResults: true, 
+                keepLongStdio: true, 
+                testResults: "${BASE_DIR}/build/junit-*.xml")
+            }
+          }
+        }
+        stage('Docker tests') {
+          agent { label 'linux && docker' }
+          environment {
+            PATH = "${env.PATH}:${env.HUDSON_HOME}/go/bin/:${env.WORKSPACE}/bin"
+            GOPATH = "${env.WORKSPACE}"
+          }
+          
+          when { 
+            beforeAgent true
+            allOf { 
+              //branch 'master';
+              environment name: 'integration_test_ci', value: 'true' 
+            }
+          }
+          steps {
+            withEnvWrapper() {
+              unstash 'source'
+              dir("${BASE_DIR}"){    
+                sh """#!/bin/bash
+                ./scripts/jenkins/docker-test.sh
+                """
+                codecov('apm-agent-go')
+              }
+            }
+          }
+          post { 
+            always { 
+              coverageReport("${BASE_DIR}/build")
+              junit(allowEmptyResults: true, 
+                keepLongStdio: true, 
+                testResults: "${BASE_DIR}/build/junit-*.xml")
+            }
+          }
+        }
+        
+        /**
+         run Go integration test with the commit version on master branch.
+        */
+        stage('Integration test master') { 
+          agent { label 'linux' }
+          when { 
+            beforeAgent true
+            allOf { 
+              //branch 'master';
+              environment name: 'integration_test_master_ci', value: 'true' 
+            }
+          }
+          steps {
```

Comment:
Is this going to run integration tests with the server and all the agents on each build of the Go agent? That sounds like it's going to be very slow. Can we make that bit periodic, rather than triggering on each build?

Reply:
no, only on master branch, on a PR it does not make sense